editor/project_manager.cpp

```diff
@@ -1751,12 +1751,16 @@ void ProjectList::erase_selected_projects(bool p_delete_project_contents) {
 	for (int i = 0; i < _projects.size(); ++i) {
 		Item &item = _projects.write[i];
 		if (_selected_project_paths.has(item.path) && item.control->is_visible()) {
-			_config.erase_section(item.path);
-
 			if (p_delete_project_contents) {
-				OS::get_singleton()->move_to_trash(item.path);
+				Error err = OS::get_singleton()->move_to_trash(item.path);
```

Comment:
This should handle other errors too, no? I.e., I'd add a check for `!= OK`. It should likely inform the user via popup that the action failed and bail out/return without erasing the section, as it was not, in fact, deleted.